src/wayland/data_device/mod.rs

```diff
@@ -71,7 +84,22 @@ pub enum DataDeviceEvent {
     /// A client has set the selection
     NewSelection(Option<Resource<wl_data_source::WlDataSource>>),
     /// A client started a drag'n'drop as response to a user pointer action
-    DnDStarted(Option<Resource<wl_data_source::WlDataSource>>),
+    DnDStarted {
+        /// The data source provided by the client
+        ///
+        /// If it is `None`, this means the DnD is restricted to surfaces of the
+        /// same client and the client will manage data transfert by itself.
+        source: Option<Resource<wl_data_source::WlDataSource>>,
+        /// The icon the client requested to be used to replace the cursor icon
+        /// during the drag'n'drop.
+        icon: Option<Resource<wl_surface::WlSurface>>,
```

Comment:
Note that the icon doesn't replace necessarily the cursor. It's usually an additional surface drawn underneath the cursor (which is usually set by the compositor depending on the current drag-and-drop action).

Reply:
Thanks for the clarification! I'll correct the wording.